src/test/tcp/test_tcp.c

```diff
@@ -49,6 +52,45 @@ static void _mylog(const char* fileName, const int lineNum, const char* funcName
     fflush(stdout);
 }
 
+static void queue_send_u16(const char* queuename, uint16_t i) {
+    mqd_t mq = mq_open(queuename, O_WRONLY|O_CREAT, 0644, NULL);
+    g_assert(mq != (mqd_t)-1);
+
+    // convert i in string before sending it
+    char buf[10] = {0};
+    int len = snprintf(buf, 10, "%d", i);
+    mq_send(mq, buf, len, 10);
+
+    mq_close(mq);
+}
+
+static short queue_recv_u16(const char* queuename) {
+    mqd_t mq;
+    guint64 val;
+    GError *error = NULL;
+    struct mq_attr attr;
+    unsigned int priorite;
+    char buf[10] = {0};
+
+    // let the server write in the queue
+    usleep(1000);
+
+    mq = mq_open(queuename, O_RDONLY);
+    g_assert(mq != (mqd_t)-1);
```

Comment:
assert (and g_assert) are generally for catching programming errors (such as a function receiving NULL that is documented as not accepting NULL). It's generally not safe to use them for catching runtime errors, since they can be disabled in some builds.

It'd also be good to output the more specific error from errno, e.g. using `perror` or `strerror`. Alternatively you can use `assert_true_errno` from https://github.com/shadow/shadow/blob/master/src/test/test_glib_helpers.h